lib/libmpa/mpa_mem_static.c

```diff
@@ -66,29 +45,54 @@ void mpa_init_scratch_mem(mpa_scratch_mem pool, int nr_vars, int max_bits)
  *  mpa_alloc_static_temp_var
  *
  */
-mpanum mpa_alloc_static_temp_var(mpanum *var, mpa_scratch_mem pool)
+mpanum mpa_alloc_static_temp_var_size(int size_bits, mpanum *var,
```

Comment:
Please add a description of the memory allocation algorithm, assumptions, limitations and strengths etc.